### PR TITLE
added force_order param in violin FF

### PR DESCRIPTION
**Pull request by Kully**
Re: https://github.com/plotly/plotly.py/issues/745

### CODE REVIEW

**Comment by Kully** [comment]
@cldougl  could you give this a review when you have a minute or two?

File: plotly/figure_factory/_violin.py [hunk 1]
```diff
@@ -435,6 +440,8 @@ def create_violin(data, data_header=None, group_header=None, colors=None,
         number and will be used to color the violin plots if a colorscale
         is being used.
     :param (bool) rugplot: determines if a rugplot is draw on violin plot.
+    :param (bool) forced_order: determines if violins are sorted
```
Comment (cldougl) [inline plotly/figure_factory/_violin.py]: I would probably call this param `sort`

File: plotly/figure_factory/_violin.py [hunk 2]
```diff
@@ -435,6 +440,8 @@ def create_violin(data, data_header=None, group_header=None, colors=None,
...
+        alphabetically (True) or by inputted order (False).
```
Comment (cldougl) [inline plotly/figure_factory/_violin.py]: input order

@Kully can you add defaults to this doc string (like https://github.com/plotly/plotly.py/blob/master/plotly/figure_factory/_distplot.py#L63-L70). It seems a little confusing to list True before False though False is the default

File: plotly/figure_factory/_violin.py [hunk 3]
```diff
@@ -627,6 +635,7 @@ def create_violin(data, data_header=None, group_header=None, colors=None,
 
             fig = violin_colorscale(
                 data, data_header, group_header, valid_colors,
-                use_colorscale, group_stats, rugplot, height, width, title
+                use_colorscale, group_stats, rugplot, forced_order, height,
```
Comment (cldougl) [inline plotly/figure_factory/_violin.py]: still forced order here

File: plotly/figure_factory/_violin.py [hunk 4]
```diff
@@ -562,8 +569,7 @@ def create_violin(data, data_header=None, group_header=None, colors=None,
             data = data[data_header].values.tolist()
 
         # call the plotting functions
-        plot_data, plot_xrange = violinplot(data, fillcolor=valid_colors[0],
-                                            rugplot=rugplot)
```
Comment (cldougl) [inline plotly/figure_factory/_violin.py]: @Kully did you want to take `rugplot` out of here?

File: plotly/figure_factory/_violin.py [hunk 5]
```diff
@@ -429,12 +434,15 @@ def create_violin(data, data_header=None, group_header=None, colors=None,
...
-        list of two rgb colors).
+        list of two rgb colors). Default = False
     :param (dict) group_stats: a dictioanry where each key is a unique
         value from the group_header column in data. Each value must be a
         number and will be used to color the violin plots if a colorscale
         is being used.
     :param (bool) rugplot: determines if a rugplot is draw on violin plot.
+        Default = True
+    :param (bool) sort: determines if violins are sorted. Default = False
+        alphabetically (True) or by inputted order (False).
```
Comment (cldougl) [inline plotly/figure_factory/_violin.py]: input instead of inputted

Comment (cldougl) [inline plotly/figure_factory/_violin.py]: @kully can you put the default at the end, looks like it got added to the middle of the sentence here.

**Comment by Kully** [comment]
@cldougl

**Comment by cldougl** [comment]
💃 💃

**Pull request merged**